tests/func/experiments/test_queue.py

```diff
@@ -0,0 +1,104 @@
+import pytest
+from funcy import first
+
+from dvc.exceptions import InvalidArgumentError
+
+
+def to_dict(tasks):
+    status_dict = {}
+    for task in tasks:
+        status_dict[task["name"]] = task["status"]
+    return status_dict
+
+
+@pytest.fixture
+def success_tasks(tmp_dir, dvc, scm, test_queue, exp_stage):
+    queue_length = 3
+    name_list = []
+    for i in range(queue_length):
+        name = f"success{i}"
+        name_list.append(name)
+        dvc.experiments.run(
+            exp_stage.addressing, params=[f"foo={i}"], queue=True, name=name
+        )
+    dvc.experiments.run(run_all=True)
+    return name_list
+
+
+@pytest.fixture
+def failed_tasks(tmp_dir, dvc, scm, test_queue, failed_exp_stage):
+    queue_length = 3
+    name_list = []
+    for i in range(queue_length):
+        name = f"failed{i}"
+        name_list.append(name)
+        dvc.experiments.run(
+            failed_exp_stage.addressing,
+            params=[f"foo={i+queue_length}"],
+            queue=True,
+            name=name,
+        )
+    dvc.experiments.run(run_all=True)
+    return name_list
+
+
+@pytest.mark.parametrize("follow", [True, False])
+def test_celery_logs(
+    tmp_dir,
+    scm,
+    dvc,
+    failed_exp_stage,
+    test_queue,
+    follow,
+    capsys,
+):
+    dvc.experiments.run(failed_exp_stage.addressing, queue=True)
+    dvc.experiments.run(run_all=True)
+
+    queue = dvc.experiments.celery_queue
+    done_result = first(queue.iter_done())
+    name = done_result.entry.stash_rev
+    captured = capsys.readouterr()
+    queue.logs(name, follow=follow)
+    captured = capsys.readouterr()
+    assert "failed to reproduce 'failed-copy-file'" in captured.out
+
+
+def test_queue_remove_done(dvc, failed_tasks, success_tasks):
```

Comment:
Do we need still need this test? With unit tests for `remove()` we should not need any additional func tests here

Reply:
I think we can simplify them. We tested the remove method case by case as well as the cli test. But we don't have a place to test the whole process. This is why I added a functional test here, But I think we can simplify it, We don't need to test all of the cases, but only one of them.